### ISSUE TITLE
ios 如果请求地址返回比较慢，会阻塞主线程，卡死

### ISSUE THREAD

**Issue by liqping** [flutterchina/dio#825]
### New Issue Checklist

- [ ] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

已经定位到引起阻塞的代码段，位于adapters/io_adapter.dart/DefaultHttpClientAdapter类中的 Future<ResponseBody> fetch函数，Future requestFuture = _httpClient.openUrl(options.method, options.uri); request = await requestFuture;，android没发现卡顿，ios 会阻塞主线程，其中的url为https://api.hylyzxw.cn/upload/upfiles 用这个url测试

**Comment by rhymelph** [comment]
I have the same problem

**Comment by xalanq** [comment]
same problem too

**Comment by xuzheng94** [comment]
I have the same problem
